### PR TITLE
[OpaquePointers] Rewrite joint_matrix tests

### PR DESCRIPTION
**Pull request by MrSidims**
This patch adds joint_matrix reverse translation to target extension type and starts rewriting all of the tests.

Some tests are being removed as outdated

Remaining tests to add after the patch:
1. tf32 test
2. element wise operations test

### CODE REVIEW

**Comment by MrSidims** [comment]
@jcranmer-intel @vmaksimo please take a look

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -450,26 +450,28 @@ Type *SPIRVToLLVM::transType(SPIRVType *T, bool UseTPT) {
...
-                                     transTypeToOCLTypeName(MT->getCompType()),
-                                     Params, !UseTPT));
-    std::string ComponentTypeName;
+      return mapType(
+          T, llvm::TargetExtType::get(*Context, "spirv.JointMatrixINTEL",
+                                      transType(MT->getCompType()), Params));
     switch (static_cast<SPIRVConstant *>(CTI)->getZExtIntValue()) {
     case internal::InternalJointMatrixCTI::TF32:
-      ComponentTypeName = "tf32";
+      Params.push_back(1);
```
Comment (vmaksimo) [inline lib/SPIRV/SPIRVReader.cpp:458]: Do we have some enum for these values?

Comment (MrSidims) [reply lib/SPIRV/SPIRVReader.cpp:458]: Oh, we do have them :) it is "internal::InternalJointMatrixCTI::"

**Comment by asudarsa** [comment]
Ah. I see that we are running into 'opaque-pointers=0' not available errors in our CI. How do we intend to work around this or tackle this?

Thanks

**Review by jcranmer-intel — APPROVED** [review]

**Pull request merged**